### ISSUE TITLE
`dipy.data.make_fetcher` test fails with Python 3

### ISSUE THREAD

**Issue by arokem** [nipy/dipy#1013]
```
D-69-91-145-227:dipy (data-tests) $nosetests dipy/data/tests/
.EE
======================================================================
ERROR: test_fetcher.test_make_fetcher
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arokem/anaconda/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/arokem/source/dipy/dipy/data/tests/test_fetcher.py", line 42, in test_make_fetcher
    sphere_fetcher()
  File "/Users/arokem/source/dipy/dipy/data/fetcher.py", line 209, in fetcher
    fetch_data(files, folder)
  File "/Users/arokem/source/dipy/dipy/data/fetcher.py", line 161, in fetch_data
    _get_file_data(fullpath, url)
  File "/Users/arokem/source/dipy/dipy/data/fetcher.py", line 123, in _get_file_data
    response_size = opener.getheader("Content-Length")
  File "/Users/arokem/anaconda/lib/python3.5/tempfile.py", line 478, in __getattr__
    a = getattr(file, name)
AttributeError: '_io.BufferedReader' object has no attribute 'getheader'
-------------------- >> begin captured stdout << ---------------------
file:///Users/arokem/source/dipy/dipy/data/files/
/Users/arokem/source/dipy/dipy/data/files/evenly_distributed_sphere_362.npz
Downloading "sphere_name" to /var/folders/jz/4k438hl91v7gvdd592ct4lpc0000gn/T/tmpxscnttud

--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_fetcher.test_fetch_data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arokem/anaconda/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/arokem/source/dipy/dipy/data/tests/test_fetcher.py", line 58, in test_fetch_data
    fetcher.fetch_data(files, tmpdir)
  File "/Users/arokem/source/dipy/dipy/data/fetcher.py", line 161, in fetch_data
    _get_file_data(fullpath, url)
  File "/Users/arokem/source/dipy/dipy/data/fetcher.py", line 123, in _get_file_data
    response_size = opener.getheader("Content-Length")
  File "/Users/arokem/anaconda/lib/python3.5/tempfile.py", line 478, in __getattr__
    a = getattr(file, name)
AttributeError: '_io.BufferedReader' object has no attribute 'getheader'
-------------------- >> begin captured stdout << ---------------------
Downloading "testfile.txt" to /var/folders/jz/4k438hl91v7gvdd592ct4lpc0000gn/T/tmpewt9lcmu

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 3 tests in 0.501s
```

**Comment by arokem** [comment]
Went under the radar until now, because #1012

**Comment by arokem** [comment]
This is resolved by #1015
